### DISCUSSION TITLE
Creating Shares - Default Values

### DISCUSSION THREAD

**Discussion by hugopvl** [directus/directus#14018] [Feature Requests]
Good Morning,

What do you think of the suggestion that we can set default values ​​for the fields of the share creation form. or even in some cases limit a certain group of users to role "share" for example. I say this because I have had some experiences with some users who make mistakes or forget and then we have a lifetime share which is not good.

![image](https://user-images.githubusercontent.com/81326296/174778902-fff87cd9-227d-4201-9818-37e474735ef7.png)

**Comment by benhaynes** [comment]
Yeah! First, we should make sure all these fields have a placeholder... not sure why it bothers me so much to see _totally_ empty fields. 😆  Here are my thoughts on defaults:

* **Name**: Would be nice to either have a placeholder here, or a dynamically generated name
* **Role**: Ideally this uses the `public` role by default, but that might not even have access to the item. We might want to run an async test when you select a role to check/warn if _nothing_ will be available. Is that possible??
* **Password**: I think this is a good one to set by default... we should generate a reasonably secure password and make sure it's visible and easily copyable (with a copy icon/button)
* **Date Start**: I don't think this should have a default... or it could be "now"
* **Date End**: I don't think this should have a default... since it might be unexpected (should have a placeholder)
* **Max Uses**: I don't think this should have a default... since it might be unexpected (should have a placeholder: `Unlimited`) 

Again, let's remove the notes that say "Leave blank for unlimited" and move those into placeholders for when the value is delete.

**Reply by hugopvl** [reply]
> Yeah! First, we should make sure all these fields have a placeholder... not sure why it bothers me so much to see _totally_ empty fields. 😆 Here are my thoughts on defaults:
> 
> * **Name**: Would be nice to either have a placeholder here, or a dynamically generated name
> * **Role**: Ideally this uses the `public` role by default, but that might not even have access to the item. We might want to run an async test when you select a role to check/warn if _nothing_ will be available. Is that possible??
> * **Password**: I think this is a good one to set by default... we should generate a reasonably secure password and make sure it's visible and easily copyable (with a copy icon/button)
> * **Date Start**: I don't think this should have a default... or it could be "now"
> * **Date End**: I don't think this should have a default... since it might be unexpected (should have a placeholder)
> * **Max Uses**: I don't think this should have a default... since it might be unexpected (should have a placeholder: `Unlimited`)
> 
> Again, let's remove the notes that say "Leave blank for unlimited" and move those into placeholders for when the value is delete.

I agree with your line of thought :) just a small detail, I think none of them should be fixed, that is, they can be changed.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
